src/navx_node.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * navx_node.cpp
+ * Runs the Kauai Labs NavX, using modified NavX library
+ * VERSION: 0.0
+ * Last changed: 2019-10-05
+ * Authors: Jude Sauve <sauve031@umn.edu>
+ * Maintainers: Nick Schatz <schat127@umn.edu>
+ * MIT License
+ * Copyright (c) 2019 UMN Robotics
+ */
+
+// ROS Libs
+#include <ros/ros.h>
+#include <sensor_msgs/Imu.h>
+
+// Native_Libs
+#include <string>
+
+// Custom_Libs
+#include "ahrs/AHRS.h"
+
+// Subscribers (inputs)
+//    update_timer (Timer)
+//      Update loop for reading / querying IMU
+
+// Publishers (outputs)
+//    imu_pub (sensor_msgs/Imu): imu/data
+//      The published imu data
+
+// Parameters (settings)
+//    frequency (double): default=50.0
+//      The update frequency of the update loop
+
+
+// ROS Node and Publishers
+ros::NodeHandle * nh;
+ros::NodeHandle * pnh;
+ros::Publisher imu_pub;
+
+// ROS Callbacks
+void update_callback(const ros::TimerEvent&);
+
+// ROS Params
+double frequency = 50.0;
+
+// Global_Vars
+AHRS* com;
+
+int main(int argc, char** argv) {
+  // Init ROS
+  ros::init(argc, argv, "navx_node");
+  nh = new ros::NodeHandle();
+  pnh = new ros::NodeHandle("~");
+
+  // Params
+  pnh->param<double>("frequency", frequency);
+
+  // Init IMU
+  com = new AHRS("/dev/ttyACM0");
+
+  // Subscribers
+  ros::Timer update_timer = nh->createTimer(ros::Duration(1.0/frequency), update_callback);
+
+  // Publishers
+  imu_pub = nh->advertise<sensor_msgs::Imu>("imu/data", 10);
+
+  // Spin
+  ros::spin();
+}
+
+void update_callback(const ros::TimerEvent&){
+  sensor_msgs::Imu msg;
+  
```

Comment:
The msg needs a header, the frame_id needs to be a param. time can come from ros::time::now(), and seq can be a global variable incremented each loop.